### PR TITLE
[testing workflows] Migrate release checks to ci.yml

### LINKED ISSUE #47532: Consolidate all e2e tests workflows into a single yml (daily, release, ci)
Move tests into project configs to run with ci.yml and deprecate the following workflows:
- smoke-test-daily.yml
- smoke-test-release.yml

### PR DESCRIPTION
**Pull request by adimoldovan**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/47532.

Move the tests from `smoke-test-release.yml` to the new CI setup and deprecate the workflow to be removed.
Added the push to tags trigger in `ci.yml`, which will be used to determine if a release-check should run.

Whenever a tag is updated and we want to run the release tests we want to use the built and archived plugin instead of the dev version. I added a new script that creates a `.wp-env.override.json` file with the plugins form the original files but the "." dir representing the current dir to use as source of woocommerce plugin will be replaced with the url to the zip file for the given tag. This way wp-env will load the release plugin instead.
I also split the `Setup Monorepo` step in install and build and only run the build step if needed (for pull_requests and push to trunk).

There will be two different types of runs: 
- `release-checks` - for any update to a tag other than `nightly`
- `nightly-checks` - for any update to the `nightly` tag. I'm thinking this will eventually replace the `daily-e2e` run type, because it also runs every day and it also provides a prebuilt plugin that we can use to test. I only configured a single job with the `nightly-checks` event, just to test the setup.

### How to test the changes in this Pull Request:

- Tested the wp-env override script locally and with [this run](https://github.com/woocommerce/woocommerce/actions/runs/9358231562/job/25759644762) which was triggered from another workflow just for testing. The workflow was then deleted. 
- The above run also gives an image of which tests were configured to run for `release-checks` run type.
- The real test will be when the first tag is created/updated, which will probably be the `nightly` tag, after this is merged.

### CODE REVIEW

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -264,7 +265,7 @@
 					],
 					"changes": [],
 					"events": [
-						"daily-e2e"
+						"daily-e2e", "release-checks"
```
Comment (rrennick) [inline plugins/woocommerce/package.json]: For consistent formatting within the file, the additions in this section should be one array element per line.

**Review by rrennick — CHANGES_REQUESTED** [review]
Nice work on this @adimoldovan. I only had the note about package.json formatting which we should fix before merging.

File: plugins/woocommerce/package.json (hunk 1 above)
Comment (adimoldovan) [reply plugins/woocommerce/package.json]: Done

**Review by rrennick — APPROVED** [review]
Looks great.

**Pull request merged**